src/components/inputs/z-searchbar/index.tsx

```diff
@@ -241,15 +261,17 @@ export class ZSearchbar {
       return null;
     }
 
-    return (
-      <z-button
-        disabled={this.preventSubmit}
-        variant={ButtonVariant.PRIMARY}
-        onClick={() => this.handleSubmit()}
-      >
-        CERCA
-      </z-button>
-    );
+    const iconProp = this.searchButtonIconOnly ? {icon: "search"} : null;
+    const buttonLabel = this.searchButtonIconOnly ? "" : "CERCA";
+    const defaultProps = {
+      disabled: this.preventSubmit,
+      variant: this.variant,
+      size: this.size,
+      onClick: () => this.handleSubmit(),
+      ...iconProp,
```

Comment:
Forse qui bastava un `icon: this.searchButtonIconOnly ? "search" : undefined` ?

Reply:
avrebbe lasciato ovunque (tranne nel caso di icon attiva)  il bottone con un `icon=""` nella z-serarchbar che mi sembrava brutto brutto.